src/types.ts

```diff
@@ -0,0 +1,12 @@
+export type IIdType = string | number;
+
+export interface IUser {
+  id?: IIdType; // TODO Ideally the model will have this
```

Comment:
These models should have an id on them. Ideally this app would save the chat and game information to a database rather than local cache on the server, and in doing so each model would naturally have an id. This id would be what is used if you wanted to add edit/delete chat message functionality or linking to a room. I've updated the "lobby" page URL to use the room id in the URL (instead of query string). As well, when iterating over these models, the ID should be the `key` not the index value.

Here is the link to the eslint rule about using indexes as keys, from there you'll be able to find discussions and blog posts galore about why this is not advised. So in place of using an index value, the next unique value on a model would be it `id`. https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md